src/main/kotlin/org/partiql/scribe/targets/trino/TrinoDialect.kt

```diff
@@ -57,71 +60,75 @@ public open class TrinoDialect : SqlDialect() {
-                asAlias = identifierSymbol((((field.name as Expr.Lit).value) as StringValue).string!!, caseSensitivity = Identifier.CaseSensitivity.INSENSITIVE)
+                expr = field.value, asAlias = identifierSymbol(
+                    (((field.name as Expr.Lit).value) as StringValue).string!!,
+                    caseSensitivity = Identifier.CaseSensitivity.INSENSITIVE
+                )
             )
         }
-        return head concat list("(SELECT ", ")") { fieldsAsItems }
+        return tail concat list("(SELECT ", ")") { fieldsAsItems }
     }
 
-    override fun visitTypeCustom(node: Type.Custom, head: SqlBlock): SqlBlock = head concat r(node.name)
+    override fun visitTypeCustom(node: Type.Custom, tail: SqlBlock): SqlBlock = tail concat node.name
 
-    override fun visitExprCall(node: Expr.Call, head: SqlBlock): SqlBlock {
+    override fun visitExprCall(node: Expr.Call, tail: SqlBlock): SqlBlock {
         return when {
             // Trino's transform function uses `->` to separate between the element variable and the element expr.
-            node.function is Identifier.Symbol && (node.function as Identifier.Symbol).symbol == "transform"-> {
-                val arrayExpr = visitExpr(node.args[0], SqlBlock.Nil)
-                val elementVar = visitExpr(node.args[1], SqlBlock.Nil)
-                val elementExpr = visitExpr(node.args[2], SqlBlock.Nil)
-                var h = head
-                h = visitIdentifier(node.function, h)
-                h = h concat "($arrayExpr, $elementVar -> $elementExpr)"
-                h
+            node.function is Identifier.Symbol && (node.function as Identifier.Symbol).symbol == "transform" -> {
+                val arrayExpr = node.args[0].sql(dialect = this)
+                val elementVar = node.args[1].sql(dialect = this)
+                val elementExpr = node.args[2].sql(dialect = this)
+                var t = tail
+                t = visitIdentifier(node.function, t)
+                t = t concat "($arrayExpr, $elementVar -> $elementExpr)"
+                t
             }
-            else -> super.visitExprCall(node, head)
+            else -> super.visitExprCall(node, tail)
         }
     }
 
-    override fun visitTypeInt2(node: Type.Int2, head: SqlBlock): SqlBlock = head concat r("SMALLINT")
+    override fun visitTypeInt2(node: Type.Int2, tail: SqlBlock): SqlBlock = tail concat "SMALLINT"
 
-    override fun visitTypeInt4(node: Type.Int4, head: SqlBlock): SqlBlock = head concat r("INT")
+    override fun visitTypeInt4(node: Type.Int4, tail: SqlBlock): SqlBlock = tail concat "INT"
 
-    override fun visitTypeInt8(node: Type.Int8, head: SqlBlock): SqlBlock = head concat r("BIGINT")
+    override fun visitTypeInt8(node: Type.Int8, tail: SqlBlock): SqlBlock = tail concat "BIGINT"
 
-    override fun visitExprUnary(node: Expr.Unary, head: SqlBlock): SqlBlock {
+    override fun visitExprUnary(node: Expr.Unary, tail: SqlBlock): SqlBlock {
```

Comment:
Probably can also remove this (like we did in RedshiftDialect + SparkDialect) since it was patched in plk 0.14.4.